src/archivist/recorder/index.test.js

```diff
@@ -0,0 +1,509 @@
+
+export const SNAPSHOTS_PATH = path.resolve(__dirname, '../../../', config.get('recorder.snapshots.storage.git.path'));
+export const VERSIONS_PATH = path.resolve(__dirname, '../../../', config.get('recorder.versions.storage.git.path'));
+
+const { expect } = chai;
+
+const MIME_TYPE = 'text/html';
+const FETCH_DATE = new Date('2000-01-01T12:00:00.000Z');
+const FETCH_DATE_LATER = new Date('2000-01-02T12:00:00.000Z');
+
+describe('Recorder', () => {
+  const SERVICE_ID = 'test_service';
+  const TYPE = 'Terms of Service';
+
+  const adaptersTypes = {
+    git: {
+      snapshots: new GitAdapter({
+        ...config.get('recorder.snapshots.storage.git'),
+        path: SNAPSHOTS_PATH,
+        fileExtension: 'html',
+      }),
+      versions: new GitAdapter({
+        ...config.get('recorder.versions.storage.git'),
+        path: VERSIONS_PATH,
+        fileExtension: 'md',
+      }),
+    },
+    mongo: {
+      snapshots: new MongoAdapter(config.get('recorder.versions.storage.mongo')),
+      versions: new MongoAdapter(config.get('recorder.snapshots.storage.mongo')),
+    },
+  };
+
+  for (const [ adapterName, adapters ] of Object.entries(adaptersTypes)) {
+    describe(adapterName, () => {
+      describe('#recordSnapshot', () => {
+        const CONTENT = '<html><h1>ToS fixture data with UTF-8 çhãràčtęrs</h1></html>';
+        let recorder;
+        let id;
+        let isFirstRecord;
+        let record;
+
+        before(async () => {
+          recorder = new Recorder({
+            versionsStorageAdapter: adapters.versions,
+            snapshotsStorageAdapter: adapters.snapshots,
+          });
+          await recorder.initialize();
+        });
+
+        after(async () => {
+          await adapters.snapshots._removeAllRecords();
+          await recorder.finalize();
+        });
+
+        context('when a required param is missing', () => {
+          after(async () => adapters.snapshots._removeAllRecords());
+
+          const validParams = {
+            serviceId: SERVICE_ID,
+            documentType: TYPE,
+            content: CONTENT,
+            fetchDate: FETCH_DATE,
+            mimeType: MIME_TYPE,
+          };
+
+          const paramsNameToExpectedTextInError = {
+            serviceId: 'service ID',
+            documentType: 'document type',
+            fetchDate: 'fetch date',
+            content: 'content',
+            mimeType: 'mime type',
+          };
+
+          Object.entries(validParams).forEach(([testedRequiredParam]) => {
+            context(`when "${testedRequiredParam}" is missing`, () => {
+              it('throws an error', async () => {
+                try {
+                  const validParamsExceptTheOneTested = Object.fromEntries(Object.entries(validParams).filter(([paramName]) => paramName != testedRequiredParam));
+
+                  await recorder.recordSnapshot(validParamsExceptTheOneTested);
+                } catch (e) {
+                  expect(e).to.be.an('error');
```

Comment:
What about this instead of try/catch? 🙂 

```js
expect(() => recorder.recordSnapshot(validParamsExceptTheOneTested)).to.throw(testedRequiredParam);

Reply:
It does not work